### PR TITLE
Storybook: Add stories for DateFormatPicker Component

### PR DESCRIPTION
**Pull request by SainathPoojary**
Part of #67165  

## What?
This PR adds Storybook stories for the `DateFormatPicker` component to improve component documentation and testability.

## Why?
- To provide clear visual documentation for the DateFormatPicker component.
- To enable interactive testing for various DateFormatPicker scenarios.
- To enhance the development experience for Block Editor components.

## Testing Instructions
1. Run `npm run storybook:dev`
2. Open Storybook at http://localhost:50240/
3. Verify the following stories are present and functioning:
   - Default rendering of the DateFormatPicker.
   - DateFormatPicker with the format set to null.
   - DateFormatPicker with defaultFormat set to empty string.
   
## Screencast 

https://github.com/user-attachments/assets/4b6126d4-136c-40ea-856f-0c7d3f09d23f

### CODE REVIEW

**Comment by SainathPoojary** [comment]
Hi @t-hamano ,

I have made the necessary changes to the PR based on the general feedback provided. Additionally, I reviewed other similar PRs with requested changes and implemented those adjustments as well. Could you kindly take a look at the PR when you have a moment?

Thank you!

https://github.com/user-attachments/assets/93440abf-01e1-45cb-8d47-bf864b8d9aed

File: packages/block-editor/src/components/date-format-picker/stories/index.story.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import DateFormatPicker from '../';
```
Comment (t-hamano) [inline packages/block-editor/src/components/date-format-picker/stories/index.story.js]: It is recommended to place it after `WordPress dependencies`. The following order is used by convention in Gutenberg:

- External dependencies
- WordPress dependencies
- Internal dependencies

File: packages/block-editor/src/components/date-format-picker/stories/index.story.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			table: {
+				type: { summary: 'string' },
+			},
+		},
+		format: {
+			control: { type: null },
+			description:
+				'The selected date format. If `null`, _Default_ is selected.',
+			table: {
+				type: { summary: 'string' },
```
Comment (t-hamano) [inline packages/block-editor/src/components/date-format-picker/stories/index.story.js]:
```suggestion
				type: { summary: 'string | null' },
```

I think the `null` type is also allowed.

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR! I only have two small suggestions, but everything works as expected 👍

**Comment by SainathPoojary** [comment]
Thank you, @t-hamano, for the review! I have implemented the changes as you suggested. 

image

**Review by t-hamano — APPROVED** [review]
LGTM! 

P.S. Thanks to your efforts, I was able to find two issues with this component that we will address in a follow-up 👍

- The browser's default border is applied to the `fieldset` element
- There is no space between the example text and the "Choose a format" control

date-format-picker

**Pull request merged**